src/sc/proofverifier.cpp

```diff
@@ -58,241 +195,349 @@ void CScProofVerifier::LoadDataForCswVerification(const CCoinsViewCache& view, c
         else
         {
             LogPrint("sc", "%s():%d - tx [%s] added to queue with %d inputs\n",
-                __func__, __LINE__, scTx.GetHash().ToString(), txMap.size());
+                __func__, __LINE__, scTx.GetHash().ToString(), cswInputProofs.size());
         }
     }
 }
 #endif
 
-bool CScProofVerifier::BatchVerify() const
+/**
+ * @brief Runs the verification for currently queued proofs.
+ * 
+ * @return true If the verification succeeded for all the proofs.
+ * @return false If the verification failed for at least one proof.
+ */
+bool CScProofVerifier::BatchVerify()
+{
+    return BatchVerifyInternal(proofQueue);
+}
+
+/**
+ * @brief Run the batch verification over a set of proofs.
+ * 
+ * @param proofs The map containing all the proofs of any kind to be verified
+ * 
+ * @return true If the verification succeeded for all the proofs.
+ * @return false If the verification failed for at least one proof.
+ */
+bool CScProofVerifier::BatchVerifyInternal(std::map</* Cert or Tx hash */ uint256, CProofVerifierItem>& proofs)
 {
-    if(verificationMode == Verification::Loose)
+    if (verificationMode == Verification::Loose)
     {
-        return true;
+        for (auto& proof : proofs)
+        {
+            proof.second.result = ProofVerificationResult::Passed;
+            return true;
+        }
     }
 
-    if (cswEnqueuedData.size() + certEnqueuedData.size() == 0)
+    if (proofs.size() == 0)
```

Comment:
Cosmetic: consider moving this check at the very start of the method

Reply:
Fixed.